src/App.jsx

```diff
@@ -1,58 +1,107 @@
 import 'bulma/css/bulma.css';
 import './App.scss';
 
+import cn from 'classnames';
+import { useState } from 'react';
+
+const SORT_FIELD_ALPHABET = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
 export const goodsFromServer = [
   'Dumplings',
   'Carrot',
   'Eggs',
-  'Ice cream',
   'Apple',
   'Bread',
   'Fish',
   'Honey',
   'Jam',
+  'Ice cream',
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortField, reverseField }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((element1, element2) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return element1[sortField] - element2[sortField];
+
+        case SORT_FIELD_ALPHABET:
+          return element1.localeCompare(element2);
+
+        default:
+          return 0;
+      }
+    });
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState('');
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
it is should be boolean state